src/aead/gcm.rs

```diff
@@ -227,6 +227,12 @@ impl Context {
     #[cfg(target_arch = "x86_64")]
     pub(super) fn is_avx2(&self, cpu_features: cpu::Features) -> bool {
         match detect_implementation(cpu_features) {
+            #[cfg(any(
+                target_arch = "aarch64",
+                target_arch = "arm",
+                target_arch = "x86_64",
+                target_arch = "x86"
+            ))]
```

Comment:
This whole function is already guarded by `#[cfg(target_arch = "x86_64")]` so this isn't needed.